examples/super_resolution/dataset.py

```diff
@@ -0,0 +1,37 @@
+from os import listdir
+from os.path import join
+
+import torch.utils.data as data
+from PIL import Image
+
+
+def is_image_file(filename):
+    return any(filename.endswith(extension) for extension in [".png", ".jpg", ".jpeg"])
+
+
+def load_img(filepath):
+    img = Image.open(filepath).convert("YCbCr")
+    y, _, _ = img.split()
+    return y
+
+
+class DatasetFromFolder(data.Dataset):
+    def __init__(self, image_dir, input_transform=None, target_transform=None):
+        super(DatasetFromFolder, self).__init__()
+        self.image_filenames = [join(image_dir, x) for x in listdir(image_dir) if is_image_file(x)]
+
+        self.input_transform = input_transform
+        self.target_transform = target_transform
+
+    def __getitem__(self, index):
+        input = load_img(self.image_filenames[index])
+        target = input.copy()
```

Comment:
Seems like in this application we are using input image as target, so we can replace BSD data by another built-in torchvision dataset instead of manually code how to download/extract the data

Reply:
Although it is working well as shown in Colab notebook